Code/RDGeneral/Dict.h

```diff
@@ -28,29 +28,61 @@ typedef std::vector<std::string> STR_VECT;
 //! \brief The \c Dict class can be used to store objects of arbitrary
 //!        type keyed by \c strings.
 //!
-//!  The actual storage is done using \c boost::any objects.
+//!  The actual storage is done using \c RDAny objects.
 //!
 class Dict {
- public:
-  typedef std::map<std::string, boost::any> DataType;
-  Dict() { _data.clear(); };
+  struct Pair {
+    std::string key;
+    RDValue val;
 
-  Dict(const Dict &other) : _data(other._data){};
+   Pair() : key(), val() {}
+   Pair(const std::string &s, const RDValue &v) : key(s), val(v) {
+   }
+  };
+  
+  typedef std::vector<Pair> DataType;
+public:
+  Dict() : _data(), _hasNonPodData(false) {  };
```

Comment:
The hasNonPodData bool is really powerful.  RDKit doesn't add boost::any data pretty much anywhere in the tree.  This flag means that a raw copy-construct ( new Dict(otherDict) ) can be a simple memcpy.